tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,20 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] lazy val hasSubclasses: Set[String] = {
+    if (linkingUnit == null) {
+      throw new IllegalArgumentException(
+          "A class emitter created without a LinkingUnit cannot emit JS classes")
```

Comment:
I saw the null check in the method right above:
``` scala
  private[javascript] lazy val linkedClassByName: Map[String, LinkedClass] = {
    if (linkingUnit == null) {
      throw new IllegalArgumentException(
          "A class emitter created without a LinkingUnit cannot emit JS classes")
    }
    linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
  }
```
Should I update that method too? And if I may, what where those deprecated constructor useful for?